File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @param {Integer} n
# @return {ListNode}
def remove_nth_from_end(head, n)
    @count=0
    @n=n
    def recurese head
        return head if head.nil?
        ret=recurese(head.next)
        @count+=1
        node=ListNode.new(head.val)
        return ret if @count==@n
        node.next=ret
        node
    end
    recurese head
end

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    if(!head.next)
        return null
        
    function removeElement(head){
        if(!head)
            return [1,null]
         let res=removeElement(head.next)
         if(res[0]==n)
              return [res[0]+1, res[1]] 
        head.next=res[1]
         return [res[0]+1, head] 
    }
    return removeElement(head, n)[1]
};